tool/src/c2/ty.rs

```diff
@@ -302,128 +337,103 @@ impl<'ccx, 'tcx: 'ccx, 'header> TyGenContext<'ccx, 'tcx, 'header> {
-                    hir::Slice::Strs(hir::StringEncoding::UnvalidatedUtf16) => {
-                        "DiplomatStrings16View"
-                    }
-                    hir::Slice::Strs(_) => "DiplomatStringsView",
-                    &_ => unreachable!("unknown AST/HIR variant"),
+            Type::Slice(ref s) => match s {
+                hir::Slice::Primitive(
+                    _,
+                    hir::PrimitiveType::Int(IntType::U8) | hir::PrimitiveType::Byte,
+                ) => "DiplomatU8View",
+                hir::Slice::Primitive(_, hir::PrimitiveType::Int(IntType::U16)) => {
+                    "DiplomatU16View"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::Int(IntType::U32)) => {
+                    "DiplomatU32View"
                 }
-                .into(),
-            ),
+                hir::Slice::Primitive(_, hir::PrimitiveType::Int(IntType::U64)) => {
+                    "DiplomatU64View"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::Int(IntType::I8)) => "DiplomatI8View",
+                hir::Slice::Primitive(_, hir::PrimitiveType::Int(IntType::I16)) => {
+                    "DiplomatI16View"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::Int(IntType::I32)) => {
+                    "DiplomatI32View"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::Int(IntType::I64)) => {
+                    "DiplomatI64View"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::IntSize(IntSizeType::Usize)) => {
+                    "DiplomatUsizeView"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::IntSize(IntSizeType::Isize)) => {
+                    "DiplomatIsizeView"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::Bool) => "DiplomatBoolView",
+                hir::Slice::Primitive(_, hir::PrimitiveType::Float(FloatType::F32)) => {
+                    "DiplomatF32View"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::Float(FloatType::F64)) => {
+                    "DiplomatF64View"
+                }
+                hir::Slice::Primitive(_, hir::PrimitiveType::Char) => "DiplomatCharView",
+                hir::Slice::Str(_, hir::StringEncoding::UnvalidatedUtf16) => "DiplomatString16View",
+                hir::Slice::Str(_, _) => "DiplomatStringView",
+                hir::Slice::Strs(hir::StringEncoding::UnvalidatedUtf16) => "DiplomatStrings16View",
+                hir::Slice::Strs(_) => "DiplomatStringsView",
+                &_ => unreachable!("unknown AST/HIR variant"),
+            }
+            .into(),
             _ => unreachable!("unknown AST/HIR variant"),
         };
-        // Todo(breaking): We can remove this requirement
-        // and users will be forced to import more types
-        if let Some(id) = id {
```

Comment:
this is the main actual behavioral change here. `Foo.h` no longer imports all impl header files for its method dependencies